### PR TITLE
Start using and adding default properties

### PR DESCRIPTION
**Pull request by externl**
This PR starts the process of updating the code to use default properties. I specifically focused on C++ and primarily on functions that used `getProperty(AsXXX)WithDefault`. I also replaced many calls to `getProperty` where we were fetching an Ice property or relying on the default of `""` or `0`.

So, not all properties have defaults set yet. I set the defaults with the following logic:

- If we documented that the property had a default, e.g. `0`, `100`, `Random`. then I used that. If the property had no documented default, or if we documented that the default relies on other properties, the os, etc. then I did not set any explicit default value.

There is still more to do after this:
- Set defaults that are currently missing.
- Update C# and Java

### CODE REVIEW

File: config/PropertyNames.xml [hunk 1]
```diff
@@ -301,64 +301,64 @@ generated from the section label.
...
+        <property name="BatchAutoFlushSize" default="1024" />
         <property name="ChangeUser" />
-        <property name="ClassGraphDepthMax" />
+        <property name="ClassGraphDepthMax" default="100" />
         <property name="ClientAccessPolicyProtocol" />
-        <property name="Compression.Level" />
+        <property name="Compression.Level" default="1" />
         <property name="Config" />
         <property name="Connection" class="connection"/>
         <property name="ConsoleListener" />
```
Comment (bernardnormier) [inline config/PropertyNames.xml:334]: The default for ConsoleListener is undocumented. It's 1 in the code (C# only).

File: config/PropertyNames.xml [hunk 2]
```diff
@@ -507,30 +508,30 @@ generated from the section label.
...
         <property name="Registry.LMDB.MapSize" />
         <property name="Registry.LMDB.Path" />
         <property name="Registry.PermissionsVerifier" class="proxy" />
-        <property name="Registry.ReplicaName" />
+        <property name="Registry.ReplicaName" default="Master" />
         <property name="Registry.RequireNodeCertCN" />
         <property name="Registry.RequireReplicaCertCN" />
         <property name="Registry.Server" class="objectadapter" />
-        <property name="Registry.SessionFilters" />
+        <property name="Registry.SessionFilters" default="0" />
```
Comment (bernardnormier) [inline config/PropertyNames.xml:534]: For all the Trace below, the default is not 0?

File: config/PropertyNames.xml [hunk 3]
```diff
@@ -301,64 +301,64 @@ generated from the section label.
     <class name="mx" prefix-only="false">
         <suffix name="GroupBy" />
         <suffix name="Map" />
-        <suffix name="RetainDetached" />
+        <suffix name="RetainDetached" default="10" />
         <suffix name="Accept" />
         <suffix name="Reject" />
     </class>
 
     <section name="Ice">
```
Comment (bernardnormier) [inline config/PropertyNames.xml:309]: There is no default for the ThreadPool properties above?

**Review by bernardnormier — APPROVED** [review]
Good start!

File: config/PropertyNames.xml (hunk 1 above)
Comment (externl) [reply config/PropertyNames.xml:334]: I opened an issue to fix the docs

File: cpp/src/Glacier2Lib/SessionHelper.cpp [hunk 4]
```diff
@@ -842,7 +842,7 @@ Glacier2::SessionFactoryHelper::createInitData()
     // plug-in has already been setup we don't want to override the
```
Comment (pepone) [inline cpp/src/Glacier2Lib/SessionHelper.cpp:842]: Seems I forgot to remove this, when moving SSL transport into the core.

File: config/PropertyNames.xml (hunk 3 above)
Comment (externl) [reply config/PropertyNames.xml:309]: There are, I just haven't gotten to them yet. IIRC they don't read them "regularly". Will look closer at the ones that are currently unset in a follow-up PR.

File: config/PropertyNames.xml [hunk 5]
```diff
@@ -301,64 +301,64 @@ generated from the section label.
...
         <property name="Admin.InstanceName" />
-        <property name="Admin.Logger.KeepLogs" />
-        <property name="Admin.Logger.KeepTraces" />
+        <property name="Admin.Logger.KeepLogs" default="100" />
+        <property name="Admin.Logger.KeepTraces" default="100" />
         <property name="Admin.Logger.Properties" />
         <property name="Admin.ServerId" />
-        <property name="BackgroundLocatorCacheUpdates"/>
+        <property name="BackgroundLocatorCacheUpdates" default="0" />
         <property name="BatchAutoFlush" deprecated="true"/>
```
[the remark this answers is not in the document]
Comment (externl) [reply config/PropertyNames.xml:326]: Which property are you referring to?

File: config/PropertyNames.xml [hunk 6]
```diff
@@ -301,64 +301,64 @@ generated from the section label.
...
-        <property name="AcceptClassCycles" />
+        <property name="AcceptClassCycles" default="0" />
         <property name="ACM.Client" deprecated="true"/>
         <property name="ACM.Server" deprecated="true"/>
         <property name="ACM" class="acm"/>
         <property name="ACM.Client" class="acm"/>
         <property name="ACM.Server" class="acm"/>
         <property name="Admin" class="objectadapter" />
-        <property name="Admin.DelayCreation" />
+        <property name="Admin.DelayCreation" default="0" />
```
[the remark this answers is not in the document]
Comment (externl) [reply config/PropertyNames.xml:317]: I don't think we should start documenting this XML file too heavily. It's too hard to keep in sync with the docs.

File: config/PropertyNames.xml (hunk 2 above)
Comment (externl) [reply config/PropertyNames.xml:534]: Probably, I haven't gotten to properties that we don't read plainly. 
For instance grepping for `Node.Trace.Activator` doesn't show anything.

```
❯ rg -F Node.Trace.Activator
config/PropertyNames.xml
504:        <property name="Node.Trace.Activator" />

scripts/IceGridUtil.py
131:            "IceGrid.Node.Trace.Activator": 0,

java/src/Ice/src/main/java/com/zeroc/IceInternal/PropertyNames.java
810:    new Property("IceGrid.Node.Trace.Activator", false, "", false, null),

csharp/src/Ice/Internal/PropertyNames.cs
784:         new(@"IceGrid.Node.Trace.Activator", false, "", false, null),

cpp/config/icegridnode.cfg
58:IceGrid.Node.Trace.Activator=3

cpp/src/Ice/PropertyNames.cpp
802:    IceInternal::Property("IceGrid.Node.Trace.Activator", false, "", false, nullptr),
```

Comment (pepone) [reply config/PropertyNames.xml:535]: https://github.com/zeroc-ice/ice/blob/c4a96d3ea88cccd593341cc86b46732c54a468a6/cpp/src/IceGrid/TraceLevels.cpp#L47

**Pull request merged**